File: task/src/test/java/dojo/BoardTest.java
package dojo;

import org.junit.Test;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

import static dojo.Cell.Empty;
import static dojo.Cell.X;
import static java.util.Arrays.asList;
import static java.util.stream.Collectors.toList;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.junit.Assert.assertThat;

public class BoardTest {

    @Test
    public void setAValueOnTheBoard() {
        List<List<Cell>> lists = IntStream.range(0, 3)
                .mapToObj(index -> new ArrayList<>(asList(Empty, Empty, Empty)))
                .collect(toList());
        Board board = new Board(lists);
        Board updatedBoard = board.set(0, 1, X);

        assertThat(updatedBoard.get(0, 1), equalTo(X));
    }
}

File: 11_final_clean_up/src/test/kotlin/dojo/BoardTest.kt
package dojo

import dojo.Cell.Empty
import dojo.Cell.X
import org.hamcrest.core.IsEqual.equalTo
import org.junit.Assert.assertThat
import org.junit.Test

class BoardTest {

    @Test
    fun setAValueOnTheBoard() {
        val lists = 0.until(3).map { List(3, { Empty }) }

        val board = Board(lists)
        val actual = board.set(0, 1, X)

        assertThat(actual[0, 1], equalTo(X))
    }
}